src/downshift.js

```diff
@@ -129,6 +129,8 @@ class Downshift extends Component {
       state.inputValue = this.props.itemToString(state.selectedItem)
     }
     this.state = state
+    this.uniqueId = generateId()
+    this.id = this.props.id || getPrefixedId('downshift', this.uniqueId)
```

Comment:
Any reason we couldn't just use a single `id` rather than have both `id` and `uniqueId`?

Reply:
No particular reason actually. I see that there's currently no relationship between `id` in props and `id` passed in `getInputProps` or `htmlFor` in `getLabelProps`. Hence I decided to keep them separate.

Reply:
Alternatively, we could derive the label's `htmlFor` and input's `id` from component's `id`,
```js
this.inputId = firstDefined(
  this.inputId,
  props.htmlFor,
  `${this.id}-input` // derive fr component id, yields 'downshift-0-input', etc
)
```

Reply:
Added test for the auto generated ids 💥

Reply:
I was thinking it'd be:

```javascript
this.inputId = firstDefined(
  this.inputId,
  props.htmlFor,
  getPrefixedId('downshift-input-', this.id)
)
```

What do you think of that?

Or.... Actually that `getPrefixedId` util is kinda uesless. Maybe we should just do what you're talking about except as: `downshift-input-${this.id}`

What do you think? I don't really care all that much, but I'm starting to think the `getPrefixedId` should be removed...

Reply:
Yeah.. We should just make it explicit using template literal (after all that's what it does), and also consistent with how we construct the id for items, i.e.

```
getItemId(index) {
  return `${this.id}-item-${index}`
}
```

With that, we'll have
```
this.id = this.props.id || `downshift-${this.uniqueId}` // e.g. downshift-0
```

and

```
this.inputId = firstDefined(
  this.inputId,
  props.htmlFor,
  `${this.id}-input` // downshift-0-input
)
```

Reply:
Let's do that!

Reply:
Done! :boom: